kolibri/plugins/coach/assets/src/views/common/LearnerExerciseReport.vue

```diff
@@ -30,23 +102,120 @@
       <li><Answer type="error" /></li>
       <li><Answer type="hint" /></li>
     </ul>
-  </div>
+  </MultiPaneLayout>
 
 </template>
 
 
 <script>
 
+  import { mapGetters, mapState } from 'vuex';
+  import ContentRenderer from 'kolibri.coreVue.components.ContentRenderer';
+  import AttemptLogList from 'kolibri.coreVue.components.AttemptLogList';
+  import InteractionList from 'kolibri.coreVue.components.InteractionList';
+  import KCheckbox from 'kolibri.coreVue.components.KCheckbox';
+  import MultiPaneLayout from 'kolibri.coreVue.components.MultiPaneLayout';
+  import CoachContentLabel from 'kolibri.coreVue.components.CoachContentLabel';
   import commonCoach from '../common';
 
   export default {
     name: 'LearnerExerciseReport',
-    components: {},
+    components: {
+      ContentRenderer,
+      AttemptLogList,
+      InteractionList,
+      KCheckbox,
+      MultiPaneLayout,
+      CoachContentLabel,
+    },
     mixins: [commonCoach],
     $trs: {},
+    data() {
+      return {
+        showCorrectAnswer: false,
+      };
+    },
+    computed: {
+      ...mapGetters(['$coreBgLight']),
+      ...mapGetters('exerciseDetail', [
+        'currentAttemptLog',
+        'currentInteraction',
+        'currentInteractionHistory',
+        'attemptLogs',
+        'attemptLogIndex',
+      ]),
+      ...mapGetters('classSummary', ['getContentStatusForLearner']),
+      ...mapState('classSummary', ['learnerMap']),
+      ...mapState('exerciseDetail', ['attemptId', 'exercise', 'interactionIndex', 'learnerId']),
+      status() {
+        return this.getContentStatusForLearner(this.exercise.content_id, this.learnerId);
+      },
+      learner() {
+        return this.learnerMap[this.learnerId];
+      },
+      // Do not pass in answerState if showCorrectAnswer is set to true
+      // answerState has a precedence over showCorrectAnswer
+      answerState() {
+        if (
+          !this.showCorrectAnswer &&
+          this.currentInteraction &&
+          this.currentInteraction.type === 'answer'
+        ) {
+          return this.currentInteraction.answer;
+        }
+        return null;
+      },
+    },
+    methods: {
+      navigateToNewAttempt(attemptLogIndex) {
+        this.showCorrectAnswer = false;
+        this.$emit('navigate', {
+          exerciseId: this.exercise.id,
+          learnerId: this.learnerId,
+          interactionIndex: 0,
+          attemptId: this.attemptLogs[attemptLogIndex].id,
+        });
+        this.$refs.multiPaneLayout.scrollMainToTop();
+      },
+      navigateToNewInteraction(interactionIndex) {
+        this.$emit('navigate', {
+          exerciseId: this.exercise.id,
+          learnerId: this.learnerId,
+          interactionIndex,
+          attemptId: this.attemptId,
+        });
+      },
+      toggleShowCorrectAnswer() {
+        this.showCorrectAnswer = !this.showCorrectAnswer;
+        this.$forceUpdate();
```

Comment:
interesting - why is this necessary? might be worth adding a comment?

Reply:
This was copied from the old exercise detail report, so I do not actually know - there was no comment there either.

Reply:
cool